torch/lib/c10d/TCPStore.hpp

```diff
@@ -61,8 +63,24 @@ class TCPStoreDaemon {
 #endif
 };
 
+class ListenThread {
+  public:
+    typedef void (*CallbackFunction)(const std::string&, const std::string&);
```

Comment:
I don't have strong preferences on this. Just wanna point out that existing code uses either dedicated struct/class or `std::function`:

1. dedicated class:

https://github.com/pytorch/pytorch/blob/19792b45dbf30b4555c4a87512e624cdd4aa6e4c/torch/csrc/distributed/rpc/request_callback.h

https://github.com/pytorch/pytorch/blob/19792b45dbf30b4555c4a87512e624cdd4aa6e4c/torch/lib/c10d/comm.hpp#L79

2. std::function (the caller can capture whatever they need in the function):

https://github.com/pytorch/pytorch/blob/19792b45dbf30b4555c4a87512e624cdd4aa6e4c/aten/src/ATen/core/ivalue_inl.h#L449

https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/CUDAFuture.h#L74

Reply:
Thanks for the note, changed to use `std::function`